flask_peewee/tests/test_rest.py

```diff
@@ -527,6 +546,66 @@ def test_delete(self):
         f_obj = FModel.get(id=self.f1.id)
         self.assertEqual(f_obj.e, None)
 
+    def test_json_filtering(self):
+        # test that filtering on JSON fields works
+        resp = self.app.get('/api/jmodel?j_field.bar=car')
+        self.assertAPIJModels(
+            resp.get_json(),
+            JModel.select().where(JModel.j_field['bar'] == 'car').order_by(JModel.id))
+
+    def test_json_edit(self):
+        self.create_test_models()
+        # test wrong pk
+        resp = self.app.put('/api/jmodel/9000/j_field', json={'new': 'value'})
+        self.assertEqual(resp.status_code, 404)
+
+        # test wrong field name
+        resp = self.app.put('/api/jmodel/9000/no_field', json={'new': 'value'})
+        self.assertEqual(resp.status_code, 404)
+
+        url = '/api/jmodel/%s' % self.j1.id
+        resp = self.app.put(url + '/j_field', json={'new': 'value'})
+        self.assertEqual(resp.status_code, 200)
+        j1 = JModel.get(JModel.id == self.j1.id)
+        assert j1.j_field == {'foo': {'star': 'far'}, 'bar': 'car', 'new': 'value'}
```

Comment:
I would have expected `assert j1.j_field == {'new': 'value'}`. If the intention is to only insert the `new` property, then it could be `put(.../j_field/new, json="value")`.

Is this a distinction between the semantics of PATCH, PUT, and POST?